modules/openxAnalyticsAdapter.js

```diff
@@ -634,7 +667,405 @@ adapterManager.registerAnalyticsAdapter({
   code: 'openx'
 });
 
+const {
+  EVENTS: { AUCTION_INIT, BID_REQUESTED, BID_RESPONSE, BID_TIMEOUT, BID_WON }
+} = CONSTANTS;
+
+const ENDPOINT = 'https://prebid.openx.net/ox/analytics';
+let auctionMap = {};
+let auctionOrder = 1; // tracks the number of auctions ran on the page
+
+function prebidAnalyticsEventHandler({eventType, args}) {
+  utils.logMessage(eventType, Object.assign({}, args));
+  switch (eventType) {
+    case AUCTION_INIT:
+      onAuctionInit(args);
+      break;
+    case BID_REQUESTED:
+      onBidRequested(args);
+      break;
+    case BID_RESPONSE:
+      onBidResponse(args);
+      break;
+    case BID_TIMEOUT:
+      onBidTimeout(args);
+      break;
+    case BID_WON:
+      onBidWon(args);
+      break;
+    case SLOT_LOADED:
+      onSlotLoadedV2(args);
+      break;
+  }
+}
+
+/*
+TODO: type Auction
```

Comment:
type? this is javascript, how are we able going to type it?

if only prebid.js was in typescript

Reply:
I meant to add jsdoc types for it